### PR TITLE
Swift UI example

### PR DESCRIPTION
**Pull request by MerinoA**
This pull request adds the basic example of the 'FetchableRequest' property wrapper in a swiftUI view.

During the implementation of this example I discovered an issue when using the property wrapper. The initial data would not render on the initial render pass.

Screen Recording 2023-05-31 at 10 32 20 AM

To remedy this behavior, the initial setting of the wrappedValue within the sink closure is wrapped 'DispatchQueue.main.async' call. Subsequent calls are not to maintain swiftUI's animations. 

Screen Recording 2023-05-31 at 10 31 19 AM

### CODE REVIEW

File: Example/iOS-Example/SwiftUIView.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        definition: Model.fetchDefinition(),
+        sortDescriptors: [
+            NSSortDescriptor(
+                key: #keyPath(Model.updatedAt),
+                ascending: false
+            ),
+        ],
+        animation: Animation.easeIn(duration: 1.0)
+    )
+
```
Comment (lickel) [inline Example/iOS-Example/SwiftUIView.swift:24]: Nit: remove this empty line

File: Example/iOS-Example/SwiftUIView.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    )
+    
+    private var models: FetchableResults<Model>
+
+    var body: some View {
+        NavigationView {
+            List(models) { model in
+                row(for: model)
+            }
+            .listStyle(PlainListStyle())
```
Comment (lickel) [inline Example/iOS-Example/SwiftUIView.swift:32]: Nit: `.listStype(.plain)`

File: Example/iOS-Example/SwiftUIView.swift [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                    Button {
+                        try? Model().save()
+                    } label: {
+                        Image(systemName: "plus")
+                    }
+            )
+        }
+    }
+    
+    static var viewController: UIHostingController<SwiftUIView> {
```
Comment (lickel) [inline Example/iOS-Example/SwiftUIView.swift:52]: Interesting... haven't seen this pattern before...
But will ask that you move this into UIKit and keep the view "pure"

File: FetchRequests/Sources/SwiftUI/FetchableRequest.swift [hunk 4]
```diff
@@ -107,17 +110,29 @@ public struct FetchableRequest<FetchedObject: FetchableObject>: DynamicProperty
         let controller = fetchController
         let binding = $wrappedValue
         let animation = self.animation
-
+        let initialLoad = $initialLoad
         subscription.value = fetchController.objectDidChange.sink { [weak controller] in
             guard let controller else {
                 return
             }
             withAnimation(animation) {
```
Comment (lickel) [inline FetchRequests/Sources/SwiftUI/FetchableRequest.swift:118]: Should probably be checking `initialLoad.wrappedValue` before starting the animation block, no?
It's also super surprising that the first pass doesn't work.

**Review by lickel — COMMENTED** [review]
I want to play around with the eventing, but also, thanks!

**Comment by lickel** [comment]
OK, tracked it down to:

```swift
    static func fetchDefinition() -> FetchDefinition<Self> {
        let dataResetTokens: [ModelClearedToken<Self>] = [
            ModelClearedToken(),
        ]

        return FetchDefinition<Self>(
            request: { completion in
                completion(fetchAll())
            },
            objectCreationToken: ModelCreationToken<Self>(),
            dataResetTokens: dataResetTokens
        )
    }
}
```

Because it does `completion(fetchAll())` there's no asynchronicity at all.
It's particular to this object.
I'd like to fix that in a different way than bouncing to main every time in the property wrapper.

**Comment by lickel** [comment]
I’ll ask that you revert the property wrapper and I’ll take a look at this in a separate PR

File: Example/iOS-Example/Model+Fetchable.swift [hunk 5]
```diff
@@ -20,7 +20,9 @@ extension FetchableObjectProtocol where Self: Model {
 
         return FetchDefinition<Self>(
             request: { completion in
-                completion(fetchAll())
+                DispatchQueue.main.async {
```
Comment (MerinoA) [inline Example/iOS-Example/Model+Fetchable.swift:23]: @lickel This will make the swiftUI example work as expected.

Comment (MerinoA) [reply Example/iOS-Example/Model+Fetchable.swift:23]: Alternatively I was able to address the issue modifying the FetchableRequest, by wrapping the fetch definition like so
``
```
 public init(
        definition: FetchDefinition<FetchedObject>,
        sortDescriptors: [NSSortDescriptor] = [],
        debounceInsertsAndReloads: Bool = true,
        animation: Animation? = nil
    ) {
        let controller = FetchedResultsController(
            definition: FetchDefinition<FetchedObject>(
                request:{ completion in
                    DispatchQueue.main.async {
                        definition.request(completion)
                    }
                },
                objectCreationToken: definition.objectCreationToken,
                creationInclusionCheck: definition.creationInclusionCheck,
                associations: definition.associations,
                dataResetTokens: definition.dataResetTokens)
            ,
            sortDescriptors: sortDescriptors,
            debounceInsertsAndReloads: debounceInsertsAndReloads
        )

        self.init(controller: controller, animation: animation)
    }
```

But I will leave that solution up to you.

Comment (lickel) [reply Example/iOS-Example/Model+Fetchable.swift:23]: yup makes sense. I think I'd still like to fix it somewhere in the framework, but thanks!

**Review by lickel — APPROVED** [review]

**Pull request merged**